flink-table/flink-table-common/src/main/java/org/apache/flink/table/connector/ChangelogMode.java

```diff
@@ -71,7 +82,21 @@ public static ChangelogMode insertOnly() {
      * contain {@link RowKind#UPDATE_BEFORE} rows.
      */
     public static ChangelogMode upsert() {
-        return UPSERT;
+        return upsert(false);
+    }
+
+    /**
+     * Shortcut for an upsert changelog that describes idempotent updates on a key and thus does not
+     * contain {@link RowKind#UPDATE_BEFORE} rows.
+     *
+     * <p>Tells the system the DELETEs contain the full record and not just the key.
+     */
+    public static ChangelogMode upsert(boolean withFullDeletes) {
+        if (withFullDeletes) {
```

Comment:
Should we invert this? Otherwise this method and the builder are inverse and might cause unintended bugs.